build/cmake/functions.cmake

```diff
@@ -15,7 +15,9 @@ if(CMAKE_GENERATOR STREQUAL "Xcode")
     # include Obj-C files when using precompiled headers with Xcode
     set(COTIRE_UNITY_SOURCE_EXCLUDE_EXTENSIONS "" CACHE STRING "wxWidgets override of cotire exclude")
 endif()
-include(cotire)                        # For precompiled header handling
+if((wxBUILD_PRECOMP STREQUAL "ON" AND CMAKE_VERSION VERSION_LESS "3.16") OR (wxBUILD_PRECOMP STREQUAL "COTIRE"))
+    include(cotire) # For precompiled header handling
+endif()
```

Comment:
I think we always need to deprecate cotire in the future, but not it's okay.

Reply:
wxWidgets supports old distributions that come with old versions of CMake. Those should also be able to benefit from PCH, that is why I kept it.